src/classes/CRLP_ApiService_TEST.cls

```diff
@@ -0,0 +1,181 @@
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group API
+* @description Unit Tests for the Customizable Rollups Api Service class
+*/
+@IsTest(IsParallel=true)
+private class CRLP_ApiService_TEST {
+
+    /**
+     * @description Validate the getRollupDefinitions returns the expected collection of CRLP_Rollup instances.
+     * Calls the mocking utilities to build a bunch of rollup definitions and filter groups/rules fo
+     */
+    @IsTest
+    private static void shouldReturnRollupDefinitionsWhenParameterIsValid() {
+        mockRollupDefinitions();
+
+        UTIL_CustomSettingsFacade.getRollupSettingsForTests(new Customizable_Rollup_Settings__c (
+            Customizable_Rollups_Enabled__c = true
+        ));
+
+        CRLP_ApiService service = new CRLP_ApiService();
+
+        Test.startTest();
+        for (CRLP_RollupProcessingOptions.RollupType rollupType : CRLP_RollupProcessingOptions.RollupType.values()) {
+            List<CRLP_Rollup> rollups = service.getRollupDefinitions(rollupType.name());
+            System.assertEquals(1, rollups.size(),
+                    'The collection of returned rollups should contain exactly 1 instance for ' + rollupType.name());
+            System.assertEquals(2, rollups[0].rollupsMdt.size(),
+                    'There should be two Rollup__mdt records associated with the CRLP_Rollup instance');
+        }
+    }
+
+    /**
+     * @description Validate the getRollupDefinitions returns an exception if the RollupType is null or invalid
+     */
+    @IsTest
+    private static void shouldThrowExceptionWhenParameterIsInvalid() {
+        CRLP_ApiService service = new CRLP_ApiService();
+
+        UTIL_CustomSettingsFacade.getRollupSettingsForTests(new Customizable_Rollup_Settings__c (
+            Customizable_Rollups_Enabled__c = true
+        ));
+
+        Test.startTest();
+        for (String param : new String[] { null, 'NonExistentType' }) {
+            Exception actualException = null;
+            try {
+                service.getRollupDefinitions(param);
+            } catch (Exception e) {
+                actualException = e;
+            }
+            System.assertNotEquals(null, actualException,
+                    'An exception should be thrown for invalid param: ' + param);
```

Comment:
Indentation is off.

Reply:
This is the default indentation for a second line (at least in IntelliJ). The style guide does not specify the exact indentation for continued lines. I think we should leave this up to the developer - as long as it meets the readability requirements.